src/arrayMethodSort.js

```diff
@@ -5,8 +5,93 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const arr = this;
+
+    // !!!bubble sort for dividing option!!!
+    if (compareFunction && arr.length > 1 && compareFunction(arr[0],
+      arr[1]) + arr[1] === arr[0]) {
+      for (let j = 0; j < arr.length; j++) {
+        for (let i = 0; i < arr.length - 1 - j; i++) {
+          if (compareFunction(arr[i], arr[i + 1]) > 0) {
+            const temp = arr[i];
+
+            arr[i] = arr[i + 1];
+
+            arr[i + 1] = temp;
+          }
+        }
+      }
+
+      return arr;
+    }
+
+    //  !!!selection sort for simple items!!!
```

Comment:
It isn't a comment. It's a working algorithm that could replace QuickSort. I can just comment QuickSort and use commented code to launch the algorithm and it will certainly work. I thought that I can try to solve this task in several approaches and they were to work. In general, they work but second or first algorithm is excessive